### DISCUSSION TITLE
Changed the layout component to client side with "use client", and vercel can not compile it(Next js 13)

### DISCUSSION THREAD

**Discussion by Michalis-Mel** [vercel/next.js#51368] [Help] [answered]
### Summary

Here is the error that vercel gives me:
> Failed to compile.
> --
> ./app/layout.jsx
> ReactServerComponentsError:
> Error: Command "npm run build" exited with 1

I tried fixing it with overriding the build command in vercel project's settings `CI='' npm run build` and `CI='' npm run build` but i kept getting the same error.

Here is the code of the layout.jsx file:
`"use client";
import localFont from "next/font/local";
import { useState } from "react";
import { motion } from "framer-motion";

import Nav from "./components/Nav/Nav";
import Loading from "./components/loading";

import "./styles/app.scss";

export default function RootLayout({ children }) {
  const [loading, setLoading] = useState(false);

  return (
    <html lang="en">
      {loading ? (
        <motion.body
          initial={{
            opacity: 0,
          }}
          animate={{
            opacity: 1,
            transition: {
              duration: 3,
              ease: "linear",
            },
          }}
          className={nunito.className}
          suppressHydrationWarning={true}
        >
          <Nav />
          {children}
        </motion.body>
      ) : (
        <body className={nunito.className} suppressHydrationWarning={true}>
          <Loading setLoading={setLoading} />
        </body>
      )}
    </html>
  );
}`

### Example

https://github.com/Michalis-Mel/final-portfolio

**Comment by icyJoseph** [comment]
Mmm this is tricky because your RootLayout is exporting metadata. I think you need to refactor, so that you can export the metadata at that level.

**Reply by icyJoseph** [reply, marked as answer]
Something like this:

```jsx
// app/layout.jsx
import "./styles/app.scss";
import { ClientRootLayout } from "./client-layout";

export const metadata = {
  title: "M² Portfolio",
  description:
    "Welcome to my space-themed portfolio! As a front-end developer specializing in Next.js 13 and SCSS, I have crafted a captivating digital universe that showcases my skills and projects. ",
  keywords: [
    "portfolio",
    "app",
    "next.js 13",
    "sass",
    "scss",
    "react",
    "space",
  ],
};

export default function RootLayout({ children }) {
  return <ClientRootLayout>{children}</ClientRootLayout>;
}
```

And then the client-layout.jsx:

```jsx
"use client";
import localFont from "next/font/local";
import { useState } from "react";
import { motion } from "framer-motion";

import Nav from "./components/Nav/Nav";
import Loading from "./components/loading";

const nunito = localFont({
  src: [
    {
      path: "./fonts/Nunito-Regular.woff2",
      weight: "400",
      style: "normal",
    },
    {
      path: "./fonts/Nunito-Medium.woff2",
      weight: "500",
      style: "normal",
    },
    {
      path: "./fonts/Nunito-Bold.woff2",
      weight: "700",
      style: "normal",
    },
  ],
});

export function ClientRootLayout({ children }) {
  const [loading, setLoading] = useState(false);

  return (
    <html lang="en">
      {loading ? (
        <motion.body
          initial={{
            opacity: 0,
          }}
          animate={{
            opacity: 1,
            transition: {
              duration: 3,
              ease: "linear",
            },
          }}
          className={nunito.className}
          suppressHydrationWarning={true}
        >
          <Nav />
          {children}
        </motion.body>
      ) : (
        <body className={nunito.className} suppressHydrationWarning={true}>
          <Loading setLoading={setLoading} />
        </body>
      )}
    </html>
  );
}

```

**Reply by Michalis-Mel** [reply]
Thank you very much. It worked!
 It is amazing that something that i have been trying to fix for hours was solved in a matter of minutes. I really hope i can get at the same level as you  some day!
